quickwit-indexing/failpoints/mod.rs

```diff
@@ -276,10 +278,7 @@ async fn test_merge_executor_controlled_directory_kill_switch() -> anyhow::Resul
     }
 
     let merge_scratch = MergeScratch {
-        merge_operation: MergeOperation {
-            merge_split_id: new_split_id(),
-            splits: split_metadatas,
-        },
+        merge_operation: MergeOperation::new_merge_operation(split_metadatas),
```

Comment:
BTW, `MergeOperation::new_merge_operation` feels redundant.

Reply:
True.

We can have:
- `MergeOperation::new(splits)` for classic merge operation
- `MergeOperation::new_delete_and_merge(split)`

WDYT?